### PR TITLE
Improve gulag logging

### PR DESCRIPTION
**Pull request by tsunyoku**
What changes:

- Logging format (now split between action and (optional) reason)
- Allows people to get the context & reason for a log without using regex

Includes a migration script to fix old logs. Looking at a merge once other branches are merged first, to avoid conflicts.

### CODE REVIEW

File: app/constants/commands.py [hunk 1]
```diff
@@ -767,13 +787,11 @@ async def addnote(ctx: Context) -> Optional[str]:
...
-    log_msg = f'{ctx.player} added note: {" ".join(ctx.args[1:])}'
-
     await app.state.services.database.execute(
         "INSERT INTO logs "
-        "(`from`, `to`, `msg`, `time`) "
-        "VALUES (:from, :to, :msg, NOW())",
-        {"from": ctx.player.id, "to": t.id, "msg": log_msg},
+        "(`from`, `to`, `action`, `msg`, `time`) "
+        "VALUES (:from, :to, 'note', :msg, NOW())",
+        {"from": ctx.player.id, "to": t.id, "msg": " ".join(ctx.args[1:])},
```
Comment (cmyui) [inline app/constants/commands.py:794]: would prefer `:action` in the sql string, and add `"action": "note"` to the values here, more readable

File: app/objects/player.py [hunk 2]
```diff
@@ -488,12 +488,11 @@ async def restrict(self, admin: "Player", reason: str) -> None:
...
 
-        log_msg = f'{admin} restricted for "{reason}".'
         await app.state.services.database.execute(
             "INSERT INTO logs "
-            "(`from`, `to`, `msg`, `time`) "
-            "VALUES (:from, :to, :msg, NOW())",
-            {"from": admin.id, "to": self.id, "msg": log_msg},
+            "(`from`, `to`, `action`, `msg`, `time`) "
+            "VALUES (:from, :to, 'restrict', :msg, NOW())",
+            {"from": admin.id, "to": self.id, "msg": reason},
```
Comment (cmyui) [inline app/objects/player.py:495]: same here with 'restrict'

File: app/objects/player.py [hunk 3]
```diff
@@ -516,12 +515,11 @@ async def unrestrict(self, admin: "Player", reason: str) -> None:
...
 
-        log_msg = f'{admin} unrestricted for "{reason}".'
         await app.state.services.database.execute(
             "INSERT INTO logs "
-            "(`from`, `to`, `msg`, `time`) "
-            "VALUES (:from, :to, :msg, NOW())",
-            {"from": admin.id, "to": self.id, "msg": log_msg},
+            "(`from`, `to`, `action`, `msg`, `time`) "
+            "VALUES (:from, :to, 'unrestrict', :msg, NOW())",
+            {"from": admin.id, "to": self.id, "msg": reason},
```
Comment (cmyui) [inline app/objects/player.py:522]: ''

File: app/objects/player.py [hunk 4]
```diff
@@ -549,12 +547,11 @@ async def silence(self, admin: "Player", duration: int, reason: str) -> None:
...
 
-        log_msg = f'{admin} silenced ({duration}s) for "{reason}".'
         await app.state.services.database.execute(
             "INSERT INTO logs "
-            "(`from`, `to`, `msg`, `time`) "
-            "VALUES (:from, :to, :msg, NOW())",
-            {"from": admin.id, "to": self.id, "msg": log_msg},
+            "(`from`, `to`, `action`, `msg`, `time`) "
+            "VALUES (:from, :to, 'silence', :msg, NOW())",
+            {"from": admin.id, "to": self.id, "msg": reason},
```
Comment (cmyui) [inline app/objects/player.py:554]: ''

File: app/objects/player.py [hunk 5]
```diff
@@ -578,12 +575,11 @@ async def unsilence(self, admin: "Player") -> None:
...
 
-        log_msg = f"{admin} unsilenced."
         await app.state.services.database.execute(
             "INSERT INTO logs "
-            "(`from`, `to`, `msg`, `time`) "
-            "VALUES (:from, :to, :msg, NOW())",
-            {"from": admin.id, "to": self.id, "msg": log_msg},
+            "(`from`, `to`, `action`, `msg`, `time`) "
+            "VALUES (:from, :to, 'unsilence', NULL, NOW())",
+            {"from": admin.id, "to": self.id},
```
Comment (cmyui) [inline app/objects/player.py:582]: ''

File: migrations/base.sql [hunk 6]
```diff
@@ -99,7 +99,8 @@ create table logs
 		primary key,
 	`from` int not null comment 'both from and to are playerids',
 	`to` int not null,
-	msg varchar(2048) charset utf8 not null,
+	`action` varchar(32) not null,
```
Comment (cmyui) [inline migrations/base.sql:102]: is action a reserved keyword?

File: tools/migrate_logs.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+            )  # now used as reason
+
+            # get all logs & change
+            print("Getting all old logs")
+            for row in await select_conn.fetch_all(f"SELECT * FROM logs"):
+                note = row["msg"]
+
+                note_match = LOG_REGEX.match(row["msg"])
+                if not note_match:
+                    continue
```
Comment (cmyui) [inline tools/migrate_logs.py:38]: perhaps we should alert the user that not all logs will be preserved? they may want to back them up if important

**Review by cmyui — CHANGES_REQUESTED** [review]
lgtm after this! nice additions :)

File: tools/migrate_logs.py (hunk 7 above)
Comment (cmyui) [reply tools/migrate_logs.py:38]: we could also save to a file (or add an option to do so)

Comment (tsunyoku) [reply tools/migrate_logs.py:38]: this is going to keep them in the db, with their old format: for that reason. i made the decision cus jkbgl for example tested the branch pre-migrator and i didn't want his already converted notes to be fucked by it

Comment (cmyui) [reply tools/migrate_logs.py:38]: ok yep awesome works for me

File: migrations/base.sql (hunk 6 above)
Comment (tsunyoku) [reply migrations/base.sql:102]: yes

File: app/objects/player.py (hunk 2 above)
Comment (tsunyoku) [reply app/objects/player.py]: solved in https://github.com/cmyui/gulag/commit/6710da3c927f7afdf628e4f7d74d97314d27adc3

File: app/objects/player.py (hunk 4 above)
Comment (tsunyoku) [reply app/objects/player.py]: solved in https://github.com/cmyui/gulag/commit/6710da3c927f7afdf628e4f7d74d97314d27adc3

File: app/objects/player.py (hunk 3 above)
Comment (tsunyoku) [reply app/objects/player.py]: solved in https://github.com/cmyui/gulag/commit/6710da3c927f7afdf628e4f7d74d97314d27adc3

File: app/objects/player.py (hunk 5 above)
Comment (tsunyoku) [reply app/objects/player.py]: solved in https://github.com/cmyui/gulag/commit/6710da3c927f7afdf628e4f7d74d97314d27adc3

File: app/constants/commands.py (hunk 1 above)
Comment (tsunyoku) [reply app/constants/commands.py]: fixed in https://github.com/cmyui/gulag/commit/d905588385693702b0247e884b763a916beb40f6

**Comment by cmyui** [comment]
cheers :)

**Pull request merged**